preview-maker.js

```diff
@@ -1,184 +1,396 @@
 
 const RENDER_CONFIG = {
-    density: 150,
+  density: 150,
 };
 
 const parser = new DOMParser();
 
+const MINIMALISTIC_TEMPLATE = Symbol();
 const REGULAR_TEMPLATE = Symbol();
 const NEW_TEMPLATE = Symbol();
 
 const templates = {
-    [REGULAR_TEMPLATE]: fs.readFileSync(`./theme-preview.svg`, `utf8`),
-    [NEW_TEMPLATE]: fs.readFileSync(`./new-preview.svg`, `utf8`),
+  [MINIMALISTIC_TEMPLATE]: fs.readFileSync(`./new-theme-preview.svg`, `utf8`),
+  [REGULAR_TEMPLATE]: fs.readFileSync(`./theme-preview.svg`, `utf8`),
+  [NEW_TEMPLATE]: fs.readFileSync(`./new-preview.svg`, `utf8`),
 };
 
 const WALLPAPERS_AMOUNT = 32;
 const wallpapers = [];
 
 for (let index = 0; index < WALLPAPERS_AMOUNT; index++) {
-    const wallpaper = fs.readFileSync(
-        `./wallpapers/${index}.jpg`,
-        `binary`,
-    );
+  const wallpaper = fs.readFileSync(`./wallpapers/${index}.jpg`, `binary`);
 
-    wallpapers.push(wallpaper);
+  wallpapers.push(wallpaper);
 }
 
-const get = (node, className, tag) => Array.from(node.getElementsByTagName(tag))
-    .filter((element) => (
-        element.getAttribute
-        && element.getAttribute(`class`) === className
-    ));
+const get = (node, className, tag) =>
+  Array.from(node.getElementsByTagName(tag)).filter(
+    (element) =>
+      element.getAttribute && element.getAttribute(`class`) === className
+  );
 
 const getElementsByClassName = (node, className) => [
-    ...get(node, className, `rect`),
-    ...get(node, className, `circle`),
-    ...get(node, className, `path`),
-    ...get(node, className, `g`),
-    ...get(node, className, `polygon`),
-    ...get(node, className, `image`),
-    ...get(node, className, `tspan`),
-    ...get(node, className, `stop`),
+  ...get(node, className, `rect`),
+  ...get(node, className, `circle`),
+  ...get(node, className, `path`),
+  ...get(node, className, `g`),
+  ...get(node, className, `polygon`),
+  ...get(node, className, `image`),
+  ...get(node, className, `tspan`),
+  ...get(node, className, `stop`),
 ];
+function rgbToHsl(rgbArr) {
+  const hsl = snejugalColor(rgbArr);
+  const hslPreRound = {
+    hue: hsl.hue,
+    saturation: hsl.saturation * 100,
+    lightness: hsl.lightness * 100,
+    alpha: hsl.alpha,
+  };
+  const result = {
+    hue: Math.round(hslPreRound.hue),
+    saturation: Math.round(hslPreRound.saturation),
+    lightness: Math.round(hslPreRound.lightness),
+    alpha: hsl.alpha,
+  };
+  return result;
+}
+function calculateAccentColor(colors) {
+  const colorsQuantity = {};
+  let max = 0;
+  let accentHue;
+
+  for (const color of colors) {
+    if (color != 0) {
+      colorsQuantity[color] = (colorsQuantity[color] ?? 0) + 1;
+
+      if (colorsQuantity[color] > max) {
+        accentHue = color;
+        max = colorsQuantity[color];
+      }
+    }
+  }
+
+  const colorsHue = colors.filter((color) => colorsQuantity[color] == max);
+
+  if (colorsHue.length > 1) {
+    let minDifference = 360;
+    for (const outerHue of colorsHue) {
```

Comment:
Here you're doing a lot of unnecessary comparisons. Just sort the array and compare `colorsHue[i]` with `colorsHue[i + 1]`, that's more readable and efficient.